### PR TITLE
Adding unit test to robot class and disable parsing URDF file

### PR DESCRIPTION
**Pull request by mhpanah**
- Created unit test to test Robot class:
The test class publishes fake odometry and pose messages and it will also listen to commended velocity to test all the methods of Robot class.
- Disabled parsing URDF file from robot class since we haven't figure out where to put the file yet.  Also, we still don't have a use for it.

### CODE REVIEW

File: nav2_robot/test/test_robot_class.cpp [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+  pose_pub_->publish(testPose_);
+}
+
+void
+TestRobotClass::publishOdom()
+{
+  testOdom_.header.frame_id = "base_footprint";
+  testOdom_.header.stamp = rclcpp::Time();
+
+  testOdom_.pose.pose.position.x = 1.1;
```
Comment (crdelsey) [inline nav2_robot/test/test_robot_class.cpp]: You should factor out this duplicated initialization code.

File: nav2_robot/test/test_robot_class.cpp [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+  }
+
+  testOdom_.twist.twist.linear.x = 1.1;
+  testOdom_.twist.twist.linear.y = 1.2;
+  testOdom_.twist.twist.linear.z = 1.3;
+  testOdom_.twist.twist.angular.x = 1.1;
+  testOdom_.twist.twist.angular.y = 1.2;
+  testOdom_.twist.twist.angular.z = 1.3;
+
+  for (int i = 0; i < 12; i++) {
```
Comment (crdelsey) [inline nav2_robot/test/test_robot_class.cpp]: This covariance initilization code could be factored out as well.

**Comment by mkhansen-intel** [comment]
@mhpanah - please rebase to master now so CI will run. Let me know if you need help. Thanks!

**Comment by mhpanah** [comment]
> @mhpanah - please rebase to master now so CI will run. Let me know if you need help. Thanks!

Done.

File: nav2_robot/test/test_robot_class.cpp [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+void
+TestRobotClass::publishPose()
+{
+  testPose_.header.frame_id = "map";
+  testPose_.header.stamp = rclcpp::Time();
+  testPose_.pose.pose.position.x = 1.1;
+  testPose_.pose.pose.position.y = 1.2;
+  testPose_.pose.pose.position.z = 1.3;
+  testPose_.pose.pose.orientation.x = 0.1;
```
[the remark this answers is not in the document]
Comment (mhpanah) [reply nav2_robot/test/test_robot_class.cpp]: Yes, of course. This was just meant to quickly initialize it with some values for testing communications.  

I can populate the fields with valid quaternion values.

File: nav2_robot/src/ros_robot.cpp [hunk 4]
```diff
@@ -97,11 +86,12 @@ RosRobot::getCurrentVelocity(nav_msgs::msg::Odometry::SharedPtr & robot_velocity
...
 }
 
-// TODO(mhpanah): modify this method name and implementation to include robot types and Serial # (ID)
+// TODO(mhpanah): modify this method name and implementation to include robot types and Serial #(ID)
 std::string
 RosRobot::getRobotName()
 {
-  return model_.getName();
+  // Temporarily just returning a string until we enable parsing URDF file.
+  return "turtlebot";
```
[the remark this answers is not in the document]
Comment (mhpanah) [reply nav2_robot/src/ros_robot.cpp]: Updated #163 to address this.

**Comment by mhpanah** [comment]
@mkhansen-intel @SteveMacenski @crdelsey Created #181 to address your comments either later today or tomorrow morning.  Will merge this now to disable the URDF file reading.

**Comment by mhpanah** [comment]
@mkhansen-intel @SteveMacenski @crdelsey Created #181 to address your comments either later today or tomorrow morning. Will merge this now to disable the URDF file reading.

**Pull request merged**